exporter/collector/config_test.go

```diff
@@ -1,100 +0,0 @@
-// Copyright 2021 OpenTelemetry Authors
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-package collector
-
-import (
-	"path"
-	"testing"
-	"time"
-
-	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
-	"go.opentelemetry.io/collector/component/componenttest"
-	"go.opentelemetry.io/collector/config"
-	"go.opentelemetry.io/collector/config/configtest"
-	"go.opentelemetry.io/collector/exporter/exporterhelper"
-)
-
-func TestLoadConfig(t *testing.T) {
-	defer SetPdataFeatureGateForTest(true)()
-	factories, err := componenttest.NopFactories()
-	assert.Nil(t, err)
-
-	factory := NewFactory()
-	factories.Exporters[typeStr] = factory
-	cfg, err := configtest.LoadConfigAndValidate(path.Join(".", "testdata", "config.yaml"), factories)
```

Comment:
One concern here, this test is not trivial because it validates the mapstructure tags on fields of certain types (I actually ran into this a few PRs back). So as an example,

- we add a new public string struct member without a tag and cut a release
- when we go to upgrade the version of this mod in collector-contrib, there would be a test failure.
- we have to fix it, rinse and repeat.

Would it make sense to move these smoke tests into the unreleased testing module along with the integration tests? That way the actual exporter doesn't depend on it, but we can still catch these issues in CI.

Reply:
That makes sense to me.  Would you like that done in this PR?

Reply:
Up to you whatever is easiest!

Reply:
I'll do it in this PR (in a separate commit so you don't have to re-review everything), since i'd like this to merge after https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/pull/253 so it doesn't disrupt that effort.

Reply:
Done, and rebased on #253.  I had to make a wrapper struct for the config, and I decided to drop queued retry and other structs not in the config.  But otherwise, I was able to keep the config parsing test intact.